### PR TITLE
Implement the initial version of Disney SSS

### PR DESCRIPTION
**Pull request by EvgeniiG**
Feel free to review, but please don't merge. :-)

### CODE REVIEW

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/Lit.hlsl [hunk 1]
```diff
@@ -119,26 +116,18 @@ void ApplyDebugToBSDFData(inout BSDFData bsdfData)
...
 
-    float t2 = thickness * thickness;
+    float3 expOneThird = exp(((-1.0 / 3.0) * thickness) * S);
 
-    // TODO: 6 exponentials is kind of expensive... Should we use a LUT instead?
-    // T = lerp(exp(-t2 * halfRcpVariance1), exp(-t2 * halfRcpVariance2), lerpWeight2)
-    float3 transmittance = exp(-t2 * halfRcpVariance1) * lerpWeight1
-                         + exp(-t2 * halfRcpVariance2) * lerpWeight2;
-    return transmittance * tintColor;
+    return 0.5 * (expOneThird + expOneThird * expOneThird * expOneThird) * surfaceAlbedo;
```
Comment (sebastienlagarde) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/Lit.hlsl]: Add more comment

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/DrawTransmittanceGraph.shader [hunk 2]
```diff
@@ -66,17 +65,11 @@ Shader "Hidden/HDRenderPipeline/DrawTransmittanceGraph"
 
             float4 Frag(Varyings input) : SV_Target
             {
-                float thickness = _ThicknessRemap.x + input.texcoord.x * (_ThicknessRemap.y - _ThicknessRemap.x);
-                float t2        = thickness * thickness;
+                float  d = (_ThicknessRemap.x + input.texcoord.x * (_ThicknessRemap.y - _ThicknessRemap.x));
+                float3 S = _ShapeParameter.rgb;
+                float3 T = 0.5 * exp(-d * S) + 0.5 * exp(-d * S * (1.0 / 3.0));
```
Comment (sebastienlagarde) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/DrawTransmittanceGraph.shader]: This function is the same than in Lit.hlsl, please create a function for it.  Move it to CommonMaterial.hlsl

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/Resources/DrawSssProfile.shader [hunk 3]
```diff
@@ -0,0 +1,79 @@
+Shader "Hidden/HDRenderPipeline/DrawSssProfile"
+{
+    Properties
+    {
+        [HideInInspector] _StdDev1("", Vector)   = (0, 0, 0, 0)
```
Comment (sebastienlagarde) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/Resources/DrawSssProfile.shader]: Remove

File: Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl [hunk 4]
```diff
@@ -19,20 +19,44 @@ float F_Schlick(float f0, float f90, float u)
...
-    return F_Schlick(f0, 1.0, u);
+    return F_Schlick(f0, 1.0, u); // sub mul mul mul sub mad
+}
+
+float F_t_Schlick(float f0, float f90, float u)
+{
+    float x     = 1.0 - u;
+    float x5    = x * x;
+    x5          = x5 * x5 * x;
+    return (1.0 - f0) - x5 * (f90 - f0); // sub mul mul mul sub sub mad
```
Comment (sebastienlagarde) [inline Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl]: We don't need this, Ft is just 1 - F_Schlick(). Just call it like that in the code

File: Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl [hunk 5]
```diff
@@ -19,20 +19,44 @@ float F_Schlick(float f0, float f90, float u)
...
 float3 F_Schlick(float3 f0, float u)
 {
-    return F_Schlick(f0, 1.0, u);
+    float x     = 1.0 - u;
+    float x5    = x * x;
+    x5          = x5 * x5 * x;
+    return f0 * (1.0 - x5) + float3(x5, x5, x5); // sub mul mul mul sub mad*3
+}
+
+float3 F_t_Schlick(float3 f0, float u)
```
Comment (sebastienlagarde) [inline Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl]: F_TransmissionSchlick

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/Lit.hlsl (hunk 1 above)
Comment (EvgeniiG) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/Lit.hlsl]: Done.

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/Lit.hlsl [hunk 6]
```diff
@@ -277,7 +273,19 @@ void EncodeIntoGBuffer( SurfaceData surfaceData,
     }
     else if (surfaceData.materialId == MATERIALID_LIT_SSS)
     {
-        outGBuffer2 = float4(surfaceData.subsurfaceRadius, surfaceData.thickness, 0.0, surfaceData.subsurfaceProfile * rcp(SSS_PROFILES_MAX - 1));
+        // Use 16 bits to encode the thickness, and up to 8 bits to encode the profile ID.
+        // We need a lot of precision to minimize banding of NdotV-weighted thickness.
+        int   ip = surfaceData.subsurfaceProfile;   // [0, 255]
+        float ft = surfaceData.thickness;           // [0, 1]
+        int   it = int(ft * ((1 << 16) - 1));       // [0, 65535]
```
Comment (EvgeniiG) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/Lit.hlsl]: Done.

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/LitData.hlsl [hunk 7]
```diff
@@ -360,13 +360,23 @@ void GetSurfaceAndBuiltinData(FragInputs input, float3 V, inout PositionInputs p
...
     GetNormalAndTangentWS(input, V, normalTS, surfaceData.normalWS, surfaceData.tangentWS);
     // Done one time for all layered - cumulate with spec occ alpha for now
-    surfaceData.specularOcclusion *= GetHorizonOcclusion(V, surfaceData.normalWS, input.worldToTangent[2].xyz, _HorizonFade);
+    surfaceData.specularOcclusion *= GetHorizonOcclusion(V, surfaceData.normalWS, interpolatedVertexNormal, _HorizonFade);
+
+    uint transmissionMode = BitFieldExtract(_TransmissionFlags, 2, 2 * surfaceData.subsurfaceProfile);
+
+    if (transmissionMode != SSS_TRSM_MODE_THIN)
+    {
+        // Convert thickness along the normal to thickness along the viewing direction.
```
Comment (EvgeniiG) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/Material/Lit/LitData.hlsl]: Done.

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/DrawTransmittanceGraph.shader (hunk 2 above)
Comment (EvgeniiG) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/DrawTransmittanceGraph.shader]: Done.

File: Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/Resources/DrawSssProfile.shader (hunk 3 above)
Comment (EvgeniiG) [inline Assets/ScriptableRenderPipeline/HDRenderPipeline/SceneSettings/Resources/DrawSssProfile.shader]: Done.

File: Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl (hunk 4 above)
Comment (EvgeniiG) [inline Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl]: Renamed, but kept the code as it is faster.

File: Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl (hunk 5 above)
Comment (EvgeniiG) [inline Assets/ScriptableRenderPipeline/ShaderLibrary/BSDF.hlsl]: Compromise: F_Transm_Schlick(). :-)

File: Assets/ScriptableRenderPipeline/ShaderLibrary/Packing.hlsl [hunk 8]
```diff
@@ -238,9 +240,61 @@ float4 UnpackQuat(float4 packedQuat)
 }
 
 //-----------------------------------------------------------------------------
-// Byte packing
+// Integer packing
 //-----------------------------------------------------------------------------
 
+// Packs an integer stored using at most 'numBits' into a [0..1] float.
+float PackInt(uint i, uint numBits)
```
Comment (sebastienlagarde) [inline Assets/ScriptableRenderPipeline/ShaderLibrary/Packing.hlsl]: Rename PackInt  => UnpackFromInt   and UnPackInt => PackToInt

**Pull request merged**